packages/cashier-v2/src/lib/PaymentAgent/lib/PaymentAgentWithdrawal/components/PaymentAgentWithdrawalReceipt/PaymentAgentReceiptDetail/PaymentAgentReceiptDetail.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { Children } from 'react';
+import { IconTypes } from '@deriv/quill-icons';
+import { Text, useDevice } from '@deriv-com/ui';
+import styles from './PaymentAgentReceiptDetail.module.scss';
+
+type TProps = {
+    action?: string;
+    children: React.ReactNode;
+    icon: IconTypes | React.ComponentType<React.SVGAttributes<SVGElement>>;
+    shouldOpenInNewTab?: boolean;
+};
+
+const PaymentAgentReceiptDetail: React.FC<TProps> = ({ action, children, icon: Icon, shouldOpenInNewTab = false }) => {
+    const { isMobile } = useDevice();
+
+    return (
+        <div className={styles.container}>
+            <div className={styles['icon-wrapper']}>
+                <Icon data-testid='dt_detail_icon' iconSize='sm' />
+            </div>
+            <div>
+                {Children.toArray(children).map((child, id, array) => {
+                    const href = action ? `${action}:${child}` : `${child}`;
+                    return (
+                        <Text key={id} size={isMobile ? 'sm' : 'xs'}>
+                            <a
+                                href={href}
+                                {...(shouldOpenInNewTab && {
+                                    rel: 'noopener noreferrer',
+                                    target: '_blank',
+                                })}
+                            >
+                                {child}
+                                {id === array.length - 1 ? '' : ', '}
```

Comment:
Shall comma be a part of a link?

also, consider something like
```
{ Children.toArray(children).map((child, id, array) => {...}).join(', ') }
```
```

```

Reply:
@dan-deriv it is
this to lines will be merged into 1:
example:
<img width="371" alt="Screenshot 2024-03-21 at 08 57 29" src="https://github.com/binary-com/deriv-app/assets/103181646/36b4528a-cbe4-436b-8b83-026393578d27">
![Screenshot 2024-03-21 at 08 57 58](https://github.com/binary-com/deriv-app/assets/103181646/5ace7d2b-e2de-4112-977b-9421fd278282)